launch/launcher.go

```diff
@@ -124,6 +146,85 @@ func (l *Launcher) env(process Process) error {
 	})
 }
 
+func (l *Launcher) execD(process Process) error {
+	execDs, err := l.buildpackFiles(process, ExecDDirName, supportsExecD)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find all exec.d executables in layers dir, '%s'", l.LayersDir)
+	}
+	for _, execD := range execDs {
+		if err := l.ExecD.ExecD(execD, l.Env); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func supportsExecD(bp Buildpack) bool {
+	if bp.API == "" {
+		return false
+	}
+	return api.MustParse(bp.API).Compare(api.MustParse("0.5")) >= 0
+}
+
+func (l *Launcher) buildpackFiles(process Process, dirName string, filters ...func(Buildpack) bool) ([]string, error) {
```

Comment:
It looks like we're using only `process.Type` in this function, so maybe it's better that `buildpackFiles` will get only it instead of `Process`.

Reply:
I kind of like like passing around the process because functions and methods can have symmetrical signatures regardless of the field(s) they read or eventually will read from the process.

Reply:
Can you please elaborate?
If this function doesn't use anything other than `process.Type`, why do we need to pass `Process`?
If things will change in the future, we can always change the function signature.
Am I missing something?